src/modules/esl-carousel/plugin/esl-carousel.plugin.ts

```diff
@@ -1,12 +1,52 @@
-import {ESLMixinElement} from '../../esl-mixin-element/ui/esl-mixin-element';
+import {ESLMixinElement} from '../../esl-mixin-element/core';
+import {bind, ready, memoize} from '../../esl-utils/decorators';
+import {evaluate} from '../../esl-utils/misc/format';
+import {ESLMediaRuleList} from '../../esl-media-query/core';
 import {ESLCarousel} from '../core/esl-carousel';
-import {ready} from '../../esl-utils/decorators/ready';
 
 /** Base mixin plugin of {@link ESLCarousel} */
-export abstract class ESLCarouselPlugin extends ESLMixinElement {
+export abstract class ESLCarouselPlugin<Config> extends ESLMixinElement {
+  /** Option to be aped if passed non object config */
+  protected static SHORT_OPTION: string = '';
+
   /** {@link ESLCarousel} host instance */
   public override $host: ESLCarousel;
 
+  /** Plugin configuration attribute value */
+  public get configValue(): string {
+    const plugin = (this.constructor as typeof ESLCarouselPlugin);
+    return this.$$attr(plugin.is) || '';
+  }
+  public set configValue(value: string) {
+    const plugin = (this.constructor as typeof ESLCarouselPlugin);
+    this.$$attr(plugin.is, value);
+  }
+
+  /** Plugin configuration query */
+  @memoize()
+  public get configQuery(): ESLMediaRuleList<Config | null> {
+    return ESLMediaRuleList.parse(this.configValue, this.$host.media, this.parseConfig);
+  }
+
+  /** Active plugin configuration object */
+  public get config(): Config {
+    return this.configQuery.value || {} as Config;
+  }
+
+  /**
+   * Parses plugin media query value term to the config object.
+   * Provides the capability to pass a config a stringified non-strict JSON or as a string (mapped to single option configuration).
+   *
+   * Uses {@link ESLCarouselPlugin.SHORT_OPTION} to map string value to the config object.
+   */
+  @bind
+  protected parseConfig(value: string): Config | null {
```

Comment:
AS I see from the return value it cannot be null
```suggestion
  protected parseConfig(value: string): Config {
```

Reply:
It could, line 44 )

Reply:
missed it :)

Reply:
missed it:)